File: src/Net/1. First Course/4. Fourth Hole/RefactoringGolf.Store/Salesman.cs
namespace RefactoringGolf.Store
{
    public class Salesman : Employee
    {
        public int CommissionPorcentage{ get; set; }

        public decimal MonthQuota{ get; private set; }

        public Salesman(string firstName, string lastName, decimal fixedSalary, int commissionPorcentage)
            : base(firstName, lastName, fixedSalary)
        {
            this.CommissionPorcentage = commissionPorcentage;
        }

        protected override decimal SalaryBenefits()
        {
            return this.MonthQuota * this.CommissionPorcentage / 100;
        }

        public void UpdateMonthQuota(decimal addQuota)
        {
            MonthQuota = MonthQuota + addQuota;
        }
    }
}

File: src/Java/1. First Course/4. Fourth Hole/src/main/refactoringgolf/store/Salesman.java
package refactoringgolf.store;

public class Salesman extends Employee {

	private int commissionPorcentage;
	private float monthQuota;

	public Salesman(String firstName, String lastName, float fixedSalary, int commissionPorcentage) {
		super(firstName, lastName, fixedSalary);
		this.commissionPorcentage = commissionPorcentage;
	}

	public void setCommissionPorcentage(int commissionPorcentage) {
		this.commissionPorcentage = commissionPorcentage;
	}

	public int getCommissionPorcentage() {
		return commissionPorcentage;
	}

	public float getMonthQuota() {
		return monthQuota;
	}

	public void updateMonthQuota(float addQuota) {
		monthQuota = monthQuota + addQuota;
	}

	protected float salaryBenefits() {
		return monthQuota * commissionPorcentage / 100;
	}
}
